test/ng/directive/ngOptionsSpec.js

```diff
@@ -922,6 +922,25 @@ describe('ngOptions', function() {
       expect(scope.selected).toEqual([20]);
       expect(element).toEqualSelectValue([20], true);
     });
+
+
+    it('should ignore array values beginning with $$', function() {
+      scope.arr = ['one', 'two', '$$hashKey'];
```

Comment:
But I just saw that the `$$hashKey` is not an array element in this case, so it's probably wrong to check for it like I did.